File: api-tester-java/src/main/java/com/revenuecat/apitesterjava/OfferingAPI.java
package com.revenuecat.apitesterjava;

import com.revenuecat.purchases.Offering;
import com.revenuecat.purchases.Package;

import java.util.List;

@SuppressWarnings({"ConstantConditions", "unused"})
final class OfferingAPI {
    static void check() {
        final Offering offering = null;

        final String identifier = offering.getIdentifier();
        final String serverDescription = offering.getServerDescription();
        final List<Package> availablePackages = offering.getAvailablePackages();

        final Package lifetime = offering.getLifetime();
        final Package annual = offering.getAnnual();
        final Package sixMonth = offering.getSixMonth();
        final Package threeMonth = offering.getThreeMonth();
        final Package twoMonth = offering.getTwoMonth();
        final Package monthly = offering.getMonthly();
        final Package weekly = offering.getWeekly();
        final Package p = offering.get("");
        final Package p2 = offering.getPackage("");
    }
}

File: api-tester-kotlin/src/main/java/com/revenuecat/apitesterkotlin/OfferingAPI.kt
package com.revenuecat.apitesterkotlin

import com.android.billingclient.api.SkuDetails
import com.revenuecat.purchases.Offering
import com.revenuecat.purchases.Package
import com.revenuecat.purchases.PackageType

@Suppress("unused", "UNUSED_VARIABLE")
private class OfferingAPI {
    fun check(offering: Offering) {
        val identifier: String = offering.identifier
        val serverDescription: String = offering.serverDescription
        val availablePackages: List<Package> = offering.availablePackages
        val lifetime: Package? = offering.lifetime
        val annual: Package? = offering.annual
        val sixMonth: Package? = offering.sixMonth
        val threeMonth: Package? = offering.threeMonth
        val twoMonth: Package? = offering.twoMonth
        val monthly: Package? = offering.monthly
        val weekly: Package? = offering.weekly
        val (_: String, _: PackageType, _: SkuDetails, _: String) = offering[""]
        val (_: String, _: PackageType, _: SkuDetails, _: String) = offering.getPackage("")
    }
}
